crp_APL/test/include/testNode/testNode.hpp

```diff
@@ -37,32 +41,59 @@ class TestNode : public rclcpp::Node
     float* globalToEgoTransform(float globalPose[3]);
 
     void calculateRoute();
-    void mapScenario();
+    void calculateSpeed();
+    void mapPath();
+    void mapObstacles();
+    void mapMovingObjects();
+    void mapDrivableSurface();
+
     void mapEgo();
 
     rclcpp::Subscription<autoware_control_msgs::msg::Control>::SharedPtr m_sub_ctrlCmd_;
     
-    rclcpp::Publisher<crp_msgs::msg::World>::SharedPtr    m_pub_world_;
-    rclcpp::Publisher<crp_msgs::msg::Scenario>::SharedPtr m_pub_scenario_;
-    rclcpp::Publisher<crp_msgs::msg::Ego>::SharedPtr      m_pub_ego_;
+    // new setup 
+    rclcpp::Publisher<autoware_perception_msgs::msg::PredictedObjects>::SharedPtr m_pub_movingObjects_;
+    rclcpp::Publisher<autoware_perception_msgs::msg::PredictedObjects>::SharedPtr m_pub_obstacles_;
+    rclcpp::Publisher<tier4_planning_msgs::msg::PathWithLaneId>::SharedPtr        m_pub_lanePath_;
+    rclcpp::Publisher<nav_msgs::msg::OccupancyGrid>::SharedPtr                    m_pub_drivableSurface_;
+
+    rclcpp::Publisher<autoware_localization_msgs::msg::KinematicState>::SharedPtr m_pub_kinematicState_;
+    rclcpp::Publisher<crp_msgs::msg::EgoStatus>::SharedPtr m_pub_egoStatus_;
+
+    rclcpp::Publisher<crp_msgs::msg::Behavior>::SharedPtr m_pub_behavior_;
+
+    // new messages
+    tier4_planning_msgs::msg::PathWithLaneId m_outPathMsg;
+    autoware_perception_msgs::msg::PredictedObjects m_movingObjectsMsg;
+    autoware_perception_msgs::msg::PredictedObjects m_obstaclesMsg;
+    nav_msgs::msg::OccupancyGrid m_drivableSurfaceMsg;
 
-    crp_msgs::msg::World m_worldMsg;
-    crp_msgs::msg::Scenario m_scenarioMsg;
-    crp_msgs::msg::Ego m_egoMsg;
+    autoware_localization_msgs::msg::KinematicState m_kinematicState;
+    crp_msgs::msg::EgoStatus m_egoStatus;
+
+    crp_msgs::msg::Behavior m_behavior;
 
     rclcpp::TimerBase::SharedPtr m_timer_;
 
     // target values coming from the vehicle
-    float m_roadWheelAngleTarget{0.0f};
-    float m_vehicleSpeedTarget{0.0f};
-    float p_wheelBase{2.7f}; // in meters
+    double m_roadWheelAngleTarget{0.0f};
+    double m_vehicleSpeedTarget{0.0f};
+    double p_wheelBase{2.7f}; // in meters
 
-    // vehicle model, state variables: [X Y Theta], input variable: road wheel angle, parameter: longitudinal speed
-    float x[3]{0.0f};
+    double m_maximumSpeedInit{20.0f}; // in mps
+    
+    double p_previewDistance = 100; // in meters
+    double m_maximumSpeed{0.0f};
+    bool m_noEnoughPointsLeft{false};

```

Comment:
maybe these params can be in a launch file ?

Reply:
done